src/CMake/FindOSPRay.cmake

```diff
@@ -175,23 +175,29 @@ else() # ospray > 1.6.1
   #message(STATUS "  OSPRAY_VERSION=${OSPRAY_VERSION}")
   #message(STATUS "  VISIT_OSPRAY_DIR=${VISIT_OSPRAY_DIR}")
 
-  # The libraries as of ospray 2.8
+  # The libraries as of ospray 3.0
+  #DDM - why do I now need lib and .so.V where it wasn't needed before?
   set(OSPRAY_LIBRARIES ospray
-                       ospray_imgui
-                       ospray_module_ispc
-                       embree3
-                       openvkl
-                       openvkl_module_cpu_device
-                       openvkl_module_cpu_device_16
-                       openvkl_module_cpu_device_8
-                       openvkl_module_cpu_device_4
-                       rkcommon
-                       tbbmalloc)
+                       libembree4.so.4
```

Comment:
Should not be listing .so.V here, those won't work for native Windows builds where there are .lib and .dll extensions.

Reply:
BTW, I am attempting builds on Linux and Windows and will report back what I discover about this particular change.

Reply:
Thanks! Yeah for whatever reason my local setup seemed to need the full lib name. I'll take them out and remove the comment about it on on line 179.

Reply:
Our SET_UP_THIRD_PARTY function is performing a `find_library` for all the listed `LIBS`.
Due to missing plain old '.so' versions of some libs in the OSPRay install, the `find_library` call is failing. No wonder you need to list the whole file name. Unfortunately if we use the full file name, would need a different list for Windows, possibly one for MacOS too.

I suggest a change to use simple `find_package` for 3.x versions of OSPRay.
Something like this:
```
    if(EXISTS ${VISIT_OSPRAY_DIR}/lib64)
        set(LIB lib64)
    else()
        set(LIB lib)
    endif()

    set(ospray_DIR ${VISIT_OSPRAY_DIR}/${LIB}/cmake/ospray-${OSPRAY_VERSION})
    find_package(ospray NO_MODULE PATHS ${ospray_DIR})
```

Would need to add logic for installing the ospray libs alongside visit, and possibly changes to a few visit targets, haven't got that far yet.

Reply:
@demarle Could you try this version of FindOSPRay.cmake and let me know if you run into problems?

[FindOSPRay.zip](https://github.com/visit-dav/visit/files/13171624/FindOSPRay.zip)

It is only installing ospray and rkcommon libs alongside VisIt, as that appears to be all that was needed (on our linux system anyways)

Reply:
My simplified find logic doesn't work on Windows, so still investigating.

Reply:
@demarle may I ask how you came up with the list of ospray libraries that were needed? It surprised me when I modified the find module to only find ospray and rkcommon that VisIt appeared to work correctly on one of our linux machines. But perhaps some system libs were found and used at runtime.

Reply:
Sure thing!

When I built it locally I scanned the OSPRay install directory and took out the libraries that I knew were not needed by apps that use ospray. Ie omitting stuff related to ospray's own examples and tests. I then just ran visit surface and volume rendering with ospray to check if I missed something important. I might have left something extra in but I'm really surpised by your finding that only rkcommon and ospray were needed.

OSPRay always uses embree and volume rendering adds the dependency on openVKL. ispcrt (ISPC Run Time) is a new dependency as of ospray 2.12 when the infrastructure for GPU started to land. sycl is a new dependency for the GPU features in 3.0.

OSPRay tends to use runtime loading (dlopen) so I suppose it could be the case that it won't complain if openVKL is missing until volume rendering is used but I would be surprised. Perhaps it was the case that bits were statically linked in? Tomorrow I'll investigate some more.

Reply:
I went ahead and modified FindOSPRay to use file(GLOB ...) to find all the libraries in the ospray/lib(64) dir for installation alongside VisIt.  More of the libraries were definitely needed on Windows.
```
   find_package(ospray ${OSPRAY_VERSION} REQUIRED
                 PATHS ${VISIT_OSPRAY_DIR}
                 PATH_SUFFIXES lib/cmake/opsray-${OSPRAY_VERSION}
                               lib64/cmake/ospray-${OSPRAY_VERSION}
                 NO_MODULE
                 NO_DEFAULT_PATH)

    if(ospray_FOUND)
        set(HAVE_LIBOSPRAY true)
        add_definitions(-DHAVE_OSPRAY)
        if(VISIT_INSTALL_THIRD_PARTY)
            # since all the libs needed for VisIt at runtime aren't
            # enumerated in the ospray targets from find_package,
            # just install all of them 
            set(LIB_SEARCH_PATH ${VISIT_OSPRAY_DIR}/lib)
            if(EXISTS ${VISIT_OSPRAY_DIR}/lib64)
                set(LIB_SEARCH_PATH ${VISIT_OSPRAY_DIR}/lib64)
            endif()
            if(NOT ospray_lib_libs)
                file(GLOB ospray_lib_libs
                     LIST_DIRECTORIES FALSE
                     ${LIB_SEARCH_PATH}/*)
            endif()
            unset(LIB_SEARCH_PATH)
            #  install libraries
            foreach(lib ${ospray_lib_libs})
                THIRD_PARTY_INSTALL_LIBRARY(${lib})
            endforeach()

            if(NOT VISIT_HEADERS_SKIP_INSTALL)
                THIRD_PARTY_INSTALL_INCLUDE(ospray ${OSPRAY_INCLUDE_DIR})
            endif()
        endif()
    endif()

```

What do you think? 

Works so far on our Linux and Windows as far as build and install.

Ospray rendering seems to be working fine on Linux, but not on Windows. Still investigating on Windows.

Reply:
I like it. Two thoughts:
1) Perhaps we could add an "exclude list" at some point for any libraries we know not to be useful to keep minimize size.
2) When OSPRay's superbuilds cmake is told to download rather than build its own dependencies it can end up with a mixture of "lib/" and "lib64/" files. If we run into that inside VisIt's build of OSPRay we might want to revisit the either or we've got above and just append to LIB_SEARCH_PATH.

Reply:
Could you make that change to FindOSPRay then on your branch?

Reply:
Yep will do.

Reply:
Done: ie, I swapped in your globbed lib find and added the line that removes the two libs which VisIt doesn't need. 

Another, probably better, approach would be to modify bv_ospray.sh line 345 or so to tell ospray 3's superbuild to turn off apps and other parts of OSPRay that VisIt doesn't need (like build_ospray_in_source still does for ospray 1.x). To the best of my knowledge visit doesn't use openimagedenoise yet so you could save some slightly more substantial compile time and install space that way.

I checked on the lib/lib64 thing and I think that is a red herring.

Reply:
Thanks for testing that out, and removing the libs we don't need.
I like the idea of modifying bv_ospray to prevent unnecessary libs/executables from being built in the first place.
I can enter a ticket to take care of this at a later time.
Are you ready for this to merge (it works on Linux)? Or do you need the Windows version to work first?